util/ranger/types.go

```diff
@@ -247,3 +248,23 @@ func formatDatum(d types.Datum, isLeftSide bool) string {
 	}
 	return fmt.Sprintf("%v", d.GetValue())
 }
+
+const emptyRangeSize = int64(unsafe.Sizeof(Range{}))
+
+// MemoryUsage return the memory usage of Range
+func (ran *Range) MemoryUsage() (sum int64) {
+	if ran == nil {
+		return
+	}
+
+	sum = emptyRangeSize +
+		int64(cap(ran.LowVal)-len(ran.LowVal))*int64(unsafe.Sizeof(types.Datum{})) +
+		int64(cap(ran.HighVal)-len(ran.HighVal))*int64(unsafe.Sizeof(types.Datum{}))
```

Comment:
It seems a little complicated, how about removing the last two parts?
Just use the formula `emptyRangeSize + LowVals.MemUsage() + HighVals.MemUsage()`.